modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/segment_edit/SegmentEdit.es.js

```diff
@@ -42,7 +47,54 @@ class SegmentEdit extends Component {
 	};
 
 	state = {
-		membersCount: this.props.initialMembersCount || 0
+		membersCount: this.props.initialMembersCount || 0,
+		membersCountLoading: false
+	};
+
+	constructor(props) {
+		super(props);
+
+		this._debouncedFetchMembersCount = debounce(
+			this._fetchMembersCount,
+			500
+		);
+	}
+
+	_fetchMembersCount = () => {
+		const formElement = document.getElementById(this.props.formId);
```

Comment:
This seems a little strange... Is it due to the fact that the form is outside of the context of react? If so, is it possible to pass into the react tree via a prop on the jsp?

Reply:
Since all of the data the request needs is in the form already, it made sense to directly submit the form. Also if we passed the data into the react tree, it would need to be passed into the request as a `FormData` type which seemed more tedious to take specific values out of the form and transform the values into a `FormData` type.

But do you think it would be more maintainable and readable by passing the specific values in?

Reply:
I was thinking of passing in the form element itself, but since you are passing in the `id` its not a huge concern. This should be fine